src/modules/azure.rs

```diff
@@ -0,0 +1,273 @@
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "subscription" => Some(Ok(subscription_name.to_string())),
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `azure`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_config_file_location(context: &Context) -> Option<PathBuf> {
+    context
+        .get_env("AZURE_CONFIG_DIR")
+        .map(PathBuf::from)
+        .or_else(|| {
+            let mut home = context.get_home()?;
+            home.push(".azure");
+            Some(home)
+        })
+}
+
+fn get_azure_subscription_id(context: &Context) -> Option<SubscriptionId> {
+    let mut config_path = get_config_file_location(context)?;
+    config_path.push("clouds.config");
+
+    let config_file = Ini::load_from_file(config_path.as_path()).ok()?;
+    let azure_cloud_section = config_file.section(Some("AzureCloud")).unwrap();
+    let current_subscription_id = azure_cloud_section.get("subscription").unwrap();
+    Some(current_subscription_id.to_string())
+}
+
+fn get_azure_subscription_name(
+    context: &Context,
+    subscription_id: &str,
+) -> Option<SubscriptionName> {
+    let mut config_path = get_config_file_location(context)?;
+    config_path.push("azureProfile.json");
+
+    if let Some(parsed_json) = parse_json(&config_path) {
+        let subscriptions = parsed_json["subscriptions"].as_array()?;
+
+        subscriptions
+            .iter()
+            .find_map(|s| find_subscription_name(s, subscription_id.to_string()))
+    } else {
+        None
+    }
+}
+
+fn parse_json(json_file_path: &Path) -> Option<JValue> {
+    let mut buffer: Vec<u8> = Vec::new();
+
+    if let Ok(json_file) = File::open(&json_file_path) {
+        let mut reader = BufReader::new(json_file);
+        reader.read_to_end(&mut buffer).ok()?;
+    } else {
+        return None;
+    }
+
+    let bytes = buffer.as_mut_slice();
+    let decodedbuffer;
+
+    if let Some([239, 187, 191]) = bytes.get(0..3) {
+        decodedbuffer = bytes.strip_prefix(&[239, 187, 191]).unwrap();
+    } else {
+        decodedbuffer = bytes;
+    }
+
+    let parsed_json: JValue = serde_json::from_slice(decodedbuffer).ok()?;
+    Some(parsed_json)
+}
+
+fn find_subscription_name(
+    subscription: &JValue,
+    current_subscription_id: SubscriptionId,
+) -> Option<SubscriptionName> {
+    let subscription_id = subscription["id"].as_str()?;
+
+    if subscription_id == current_subscription_id {
+        let subscription_name = subscription["name"].as_str()?;
```

Comment:
Please use `.get()` instead of `[]` for indexing to avoid potential panics, and add a test for missing `id`/`name`.